taxonomy/api.py

```diff
@@ -1,252 +1,54 @@
-# pylint: disable=invalid-name
-@transaction.atomic
-def remove_term_from_learning_resource(learning_resource_id, term_id):
-    """
-    Remove Term from LearningResource.
-
-    Remove existing Term with id ``term_id`` from a LearningResource with id
-    ``learning_resource_id``.
-
-    Args:
-        learning_resource_id (int): Id for LearningResource
-        term_id (int): Id for Term
-    """
-    term = get_object_or_404(Term, id=term_id)
-    learning_resource = get_object_or_404(LearningResource,
-                                          id=learning_resource_id)
-    term.learning_resources.remove(learning_resource)
-
-
-def get_vocabularies_for_type(learning_resource_type):
-    """
-    Get vocabularies supporting the given learning object type.
-
-    Vocabularies are ordered by ``name``.
-
-    Args:
-        learning_resource_type (unicode): The learning resource type
-
-    Returns:
-        sequence of Vocabulary objects supporting the learning resource type
-    """
-    return get_object_or_404(
-        LearningResourceType,
-        name=learning_resource_type
-    ).vocabularies.order_by('name')
-
-
-def get_types_for_vocabulary(vocabulary_id):
-    """
-    Get learning object types supported by a Vocabulary.
-
-    Args:
-        vocabulary_id (int): Id for existing Vocabulary
-
-    Returns:
-        list (list of unicode): sequence of learning object types supported by
-        a LearningResource
-    """
-    learning_resource_types = get_object_or_404(
-        Vocabulary, id=vocabulary_id
-    ).learning_resource_types.order_by('name')
-
-    return [x.name for x in learning_resource_types]
-
-
-@transaction.atomic
-def add_type_for_vocabulary(learning_resource_type,
-                            vocabulary_id):
-    """
-    Add learning object type to an existing Vocabulary.
-
-    Args:
-        learning_resource_type (unicode): A learning object type
-        vocabulary_id (int): Id for existing Vocabulary
-    """
-    vocabulary = get_object_or_404(Vocabulary, id=vocabulary_id)
-    learning_resource_type = get_object_or_404(LearningResourceType,
-                                               name=learning_resource_type)
-    vocabulary.learning_resource_types.add(learning_resource_type)
-
-
-@transaction.atomic
-def remove_type_from_vocabulary(learning_resource_type_name,
-                                vocabulary_id):
-    """
-    Remove learning object type from existing Vocabulary.
-
-    Args:
-        learning_resource_type (unicode): A learning object type
-        vocabulary_id (int): Id for existing Vocabulary
-    """
-    vocabulary = get_object_or_404(Vocabulary, id=vocabulary_id)
-    learning_resource_type = get_object_or_404(
-        LearningResourceType,
-        name=learning_resource_type_name
-    )
-    vocabulary.learning_resource_types.remove(learning_resource_type)
+    repo = get_repo(repo_slug, user_id)
+    try:
+        return repo.vocabulary_set.get(
```

Comment:
Some comment here about the `DoesNotExist` thing. I won't repeat it if I see it later on in the code.